### PR TITLE
[Content Manager]- types for routing middleware

### PR DESCRIPTION
**Pull request by jhoward1994**
### What does it do?

Add types to the routing middleware in the content manager

### CODE REVIEW

File: packages/core/content-manager/server/src/middlewares/routing.ts [hunk 1]
```diff
@@ -20,18 +31,19 @@ export default async (ctx: any, next: any) => {
   const [, action] = route.handler.split('.');
 
   const configPath =
-    ct.plugin === 'admin'
+    !ct.plugin || ct.plugin === 'admin'
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/src/middlewares/routing.ts:34]: why this change is necesssary?

Comment (jhoward1994) [reply packages/core/content-manager/server/src/middlewares/routing.ts]: thanks, you're right

File: packages/core/content-manager/server/src/middlewares/routing.ts [hunk 2]
```diff
@@ -1,17 +1,28 @@
...
+
+  const ct: ContentType = strapi.contentTypes[model];
 
   if (!ct) {
     return ctx.send({ error: 'contentType.notFound' }, 404);
   }
 
-  const target = ct.plugin === 'admin' ? strapi.admin : strapi.plugin(ct.plugin);
+  let target;
+  if (!ct.plugin || ct.plugin === 'admin') {
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/src/middlewares/routing.ts:19]: similar to the other question, is this because of type checking?

Comment (jhoward1994) [reply packages/core/content-manager/server/src/middlewares/routing.ts:19]: Yes, without this we get a TS error at L22 about the input to `strapi.plugin`

**Review by Marc-Roig — APPROVED** [review]
